### DISCUSSION TITLE
How to set up authentication flow with step up and alternative passwordless authentication?

### DISCUSSION THREAD

**Discussion by sventorben** [keycloak/keycloak#26735] [Q&A]
I am trying to set up an authentication flow that can handle the following use case:

When a client requests authentication with LoA = 1 the user should be able to authenticate either via username/password or username/passwordless authentication.

I am thinking about configuring a flow like this:

- Cookie (Alternative)
- Subflow (Alternative)
  - Username Form (Required)
  - Subflow (Conditional)
    - Conditional (LoA = 1)
    - Password Form (Alternative)
    - WebAuthn Passwordless Authenticator (Alternative)
   

When a client requests authentication with LoA = 2 the user must also provide a second factor unless the initial login was via WebAuthn Passwordless Authenticator. So, I came up with this.

- Cookie (Alternative)
- Subflow (Alternative)
  - Username Form (Required)
  - Subflow (Conditional)
    - Conditional (LoA = 1)
    - Password Form (Alternative)
    - WebAuthn Passwordless Authenticator (Alternative)
  - Subflow (Conditional)
    - Conditional (LoA = 2)
    - OTP (Alternative)
    - WebAuthn Passwordless Authenticator (Alternative)

But now the user is asked for passwordless authentication twice. How to avoid this?

Basically I want the following:

- if requested LoA = 1
  - present two alternatives (password and passwordless)
  - if user chooses password, grant access
  - else if user  chooses passwordless, grant access
- if requested LoA = 2
  - present two alternatives (password and passwordless)
  - if user  chooses passwordless, grant access
  - else if user chooses password
    - ask for password
    - present two alternatives (OTP or passwordless)

Is it possible to set this up?

**Comment by hardenchant** [comment]
Hello. Did you have success on that?
I need the same thing
